projects/plugins/jetpack/modules/woocommerce-analytics/classes/class-jetpack-woocommerce-analytics-universal.php

```diff
@@ -237,30 +237,57 @@ function ( $payment_gateway ) {
 				$product->get_id()
 			);
 
-			if ( true === $include_express_payment ) {
-				wc_enqueue_js(
-					"
-					// wcpay.payment-request.availability event gets fired twice.
-					// make sure we push only one event.
-					var cartItem_{$cart_item_key}_logged = false;
-				    wp.hooks.addAction('wcpay.payment-request.availability', 'wcpay', function(args) {
-				        if ( true === cartItem_{$cart_item_key}_logged ) {
-				            return;
-				        }
-				        var properties = {$properties};
-				        properties.express_checkout = args.paymentRequestType;
-				        _wca.push(properties);
-						cartItem_{$cart_item_key}_logged = true;
-				    });
+			wc_enqueue_js(
 				"
-				);
-			} else {
-				$this->record_event(
-					'woocommerceanalytics_product_checkout',
-					$data,
-					$product->get_id()
-				);
-			}
+				var cartItem_{$cart_item_key}_logged = false;
+				var properties = {$properties};
+				// Check if jQuery is available
+				if ( typeof jQuery !== 'undefined' ) {
+					// This is only triggered on the checkout shortcode.
+					jQuery( document.body ).on( 'init_checkout', function () {
+						if ( true === cartItem_{$cart_item_key}_logged ) {
+							return;
+						}
+						wp.hooks.addAction( 'wcpay.payment-request.availability', 'wcpay', function ( args ) {
```

Comment:
`wp.hooks` is guaranteed to be available. If it's not available, this blocks a lot of JS on the checkout from working, including coupons, "Click here to login", etc. See [here](https://wordpress.org/support/topic/woocommerce-checkout-coupon-cant-add/).

Reply:
Thanks for raising this @jjcoffee - I agree we should add a check to ensure it's available before using.

Are you able to provide some reproduction steps for this issue?

Reply:
@jjcoffee - I have opened https://github.com/Automattic/jetpack/pull/36560 to solve this.